### ISSUE TITLE
Siamese network do not converge, while with raw tf the same network converges.

### ISSUE THREAD

**Issue by Aelphy** [keras-team/keras#8629]
I created a siamese like network.

When training with keras loss it does not converge at all.

With raw tensorflow loss function and training_op it converges quickly.

https://gist.github.com/Aelphy/c73aa56bf2f410b1423c63bd7087142e

**Comment by Light--** [comment]
have you found the reason?

**Comment by Aelphy** [comment]
No, just switched to raw version

**Comment by Light--** [comment]
> raw version

what do you mean by 'raw version'? where can i find the code?
